src/clj/puppetlabs/services/master/master_core.clj

```diff
@@ -937,26 +940,66 @@
                                                      "code_id"
                                                      (current-code-id-fn (get request-options "environment")))))})))
 
+(defn parse-project-compile-data
+  "Parse data required to compile a catalog inside a project. Data required includes
+   * Root path to project
+   * modulepath
+   * hiera config
+   * project_name"
+  [request-options
+   versioned-project
+   bolt-projects-dir
+   multithreaded]
+  (let [project-root (file-serving/get-project-root bolt-projects-dir versioned-project)
+        project-config (file-serving/read-bolt-project-config project-root)]
+    (-> request-options
```

Comment:
Given there is no logic around whether or not to actually add a KV to the map, why not just do a single assoc? I feel like multiple with the threading macro is useful when you conditionally want to add a pair, but otherwise is just kind of redundant.

Reply:
Good point, I had gone straight to threading but you're right that it makes more sense to just use the one assoc.